comid/integregisters.go

```diff
@@ -0,0 +1,152 @@
+	reg := make(map[IRegisterIndex]Digests)
+	return &IntegReg{Register: reg}, nil
+}
+
+// AddDigests allows inserting a series of Digests at a specific RegisterIndex
+func (i *IntegReg) AddDigests(index IRegisterIndex, digests Digests) error {
+	if len(digests) == 0 {
+		return fmt.Errorf("no digests to add")
+	}
+	for _, digest := range digests {
+		if err := i.AddDigest(index, digest); err != nil {
+			return fmt.Errorf("unable to add Digest: %w", err)
+		}
+	}
+	return nil
+}
+
+// AddDigest allows inserting a Digest at a specific RegisterIndex
+func (i *IntegReg) AddDigest(index IRegisterIndex, digest swid.HashEntry) error {
+
+	if i.Register == nil {
+		return fmt.Errorf("no register to add digest")
+	}
+	switch t := index.(type) {
+	case string, uint, uint64:
+		i.Register[t] = append(i.Register[t], digest)
+	default:
+		return fmt.Errorf("unexpected type for index: %T", t)
+	}
+	return nil
+}
+
+// RemoveIndex removes an Index from the Register
+func (i *IntegReg) RemoveIndex(index IRegisterIndex) error {
+	if i.Register == nil {
+		return fmt.Errorf("cannot remove index, non existent register")
+	}
+	switch t := index.(type) {
+	case uint, uint64, string:
+		delete(i.Register, index)
+	default:
+		return fmt.Errorf("unknown type %T for index-type-choice", t)
+	}
+	return nil
+}
+
+func (i IntegReg) MarshalCBOR() ([]byte, error) {
+	return em.Marshal(i.Register)
+}
+
+func (i *IntegReg) UnMarshalCBOR(data []byte) error {
+	return dm.Unmarshal(data, &i.Register)
+}
+
+func (i IntegReg) ToJSON() ([]byte, error) {
+	jmap := make(map[string]json.RawMessage)
+	for key, val := range i.Register {
+		var v tnv
+		switch t := key.(type) {
+		case uint, uint64:
+			b, err := json.Marshal(key)
+			v = tnv{Type: "uint", Value: b}
+			if err != nil {
+				return nil, err
+			}
+		case string:
+			b, err := json.Marshal(key)
+			v = tnv{Type: "text", Value: b}
+			if err != nil {
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unknown type %T for index-type-choice", t)
+		}
+		newk, err := json.Marshal(v)
+		if err != nil {
+			return nil, err
+		}
+		newkey := string(newk)
+		newval, err := json.Marshal(val)
+		if err != nil {
+			return nil, err
+		}
+		jmap[newkey] = newval
+	}
+	return json.Marshal(jmap)
+}
+
+func (i *IntegReg) FromJSON(data []byte) error {
+	jmap := make(map[string]json.RawMessage)
+	if err := json.Unmarshal(data, &jmap); err != nil {
+		return fmt.Errorf("register map decoding failure: %w", err)
+	}
+	for key, val := range jmap {
+		index, err := unMarshalIndex(key)
+		if err != nil {
+			return fmt.Errorf("unable to unMarshal index: %w", err)
+		}
+		var d Digests
+
+		if err = json.Unmarshal(val, &d); err != nil {
+			return fmt.Errorf("unable to UnMarshal Digests: %w", err)
+		}
+		if err = i.AddDigests(index, d); err != nil {
+			return fmt.Errorf("unable to insert register into register set: %w", err)
+		}
+	}
+	return nil
+}
+
+func unMarshalIndex(key string) (IRegisterIndex, error) {
```

Comment:
unmarshal is a single word, camel case does not apply:

```suggestion
func unmarshalIndex(key string) (IRegisterIndex, error) {
```